### DISCUSSION TITLE
CLI Tools & Applications showing as missing

### DISCUSSION THREAD

**Discussion by CharlesOural** [pimcore/pimcore#9277] [Q&A / Support]
# Bug Report

I install a brand new version of Pimcore X on a debian 10 server with php8.0 and nginx.

## But when I go to system-requirements-check (admin-->tools-->system info and tools) I got this :

![Screenshot from 2021-05-30 13-23-00](https://user-images.githubusercontent.com/7414043/120102217-353f1780-c14a-11eb-95fe-57eb079bb5bf.png)

## I check the symphony console and all tools seems to be Ok : 

![Screenshot from 2021-05-30 13-25-33](https://user-images.githubusercontent.com/7414043/120102281-87803880-c14a-11eb-915e-c00250198d07.png)

I track down the issue to pimcore/lib/Tool/Console.php and the fonction getExecutable, the variable $path is not set correctly...
When I debug the code with : 

        $paths = [];
        if (!empty($systemConfig['path_variable'])) {
            Logger::debug("Using systemconfig paths variables !");
            $paths = explode(PATH_SEPARATOR, $systemConfig['path_variable']);
        } 
        $string_debug = implode(PATH_SEPARATOR, $paths);
        Logger::debug("This is how php sees paths : " . $string_debug);
        array_push($paths, '');

And I get,

       DEBUG : This is how php sees paths : 

So I hard coded : 

        $paths = [];
        if (!empty($systemConfig['path_variable'])) {
            Logger::debug("Using systemconfig paths variables !");
            $paths = explode(PATH_SEPARATOR, $systemConfig['path_variable']);
        } 
        $paths = ["/usr/local/bin", "/usr/bin", "/bin"];
        $string_debug = implode(PATH_SEPARATOR, $paths);
        Logger::debug("This is how php sees paths : " . $string_debug);
        array_push($paths, '');

## And then all is working : 

![Screenshot from 2021-05-30 13-38-23](https://user-images.githubusercontent.com/7414043/120102655-5274e580-c14c-11eb-8750-aab558707f45.png)

I don't really understand where this bug can come from.... I reinstall twice pimcore and it is still not working....

BTW I used this install script : https://github.com/charles200000/Hoff_ProductionServers/blob/8278a37a1765e53873b225cd70c5bc1ef3f55ce5/PIMCore/InstallScripts/ProdServer_PIMCore_App.sh

Thanks for your help !

**Comment by RuudvdvNB** [comment]
Experiencing the same issue on Pimcore 6.9.4.

Paths config `$systemConfig['path_variable']` in local development returns `null` but on our testing/production servers this returns an empty array.

**Comment by RuudvdvNB** [comment]
@brusch why is this changed to a discussion? This is causing errors when using specific functions in Pimcore admin because it can't find a PHP CLI executable even tho this is actually available? In older Pimcore versions this is working in the same environments.
